python/triton/compiler/code_generator.py

```diff
@@ -425,6 +428,25 @@ def name_lookup(name: str) -> Any:
 
         return name_lookup
 
+    @contextlib.contextmanager
+    def _name_loc_prefix(self, prefix):
+        self.name_loc_as_prefix = prefix
+        yield
+        self.name_loc_as_prefix = None
+
+    def _maybe_set_loc_to_name(self, val, name):
+        from ..experimental.gluon.language._core import shared_memory_descriptor
+        from ..experimental.gluon.language.nvidia import blackwell, hopper
+        if isinstance(val, (ir.value, ir.block_argument)):
+            handle = val
+        elif _is_triton_tensor(val) or isinstance(val,
+                                                  (language.core.tensor_descriptor, shared_memory_descriptor,
+                                                   blackwell.tensor_memory_descriptor, hopper.tma.tensor_descriptor)):
+            handle = val.handle
```

Comment:
```suggestion
        elif _is_triton_value(val):
            handles = _flatten_values_to_ir([val])
```

This will associate all handles in any value to the name of the top-level value. At some point we might want more fine-grained info like `name.attr` but I think this is good enough as a start.